### DISCUSSION TITLE
Page/Layout script transition

### DISCUSSION THREAD

**Discussion by ilyaDegtyarenko** [nuxt/nuxt#17430] [Questions] [answered]
How to use js transition animation (ex: gsap) for page and layout?

**Comment by Flanker72** [comment, marked as answer]
The official recommendation is to pass settings to the `transition` component within the page wrapper: https://v3.nuxtjs.org/guide/directory-structure/pages#layouttransition-and-pagetransition

You can use `definePageMeta` like this:
```
<script setup lang="ts">
definePageMeta({
  pageTransition: {
    name: "slide-fade",
  },
  layoutTransition: {
    name: "fade",
  },
});
</script>
```

The most comprehensive docs about how Vue3 transitions works is here: https://vuejs.org/guide/built-ins/transition.html

**Reply by ilyaDegtyarenko** [reply]
You are right, thank you
And how can I set page transition default? I tried to do it in nuxt.config.ts, but it doesn't work

**Reply by ilyaDegtyarenko** [reply]
something like that

```
    pageTransition: {
        onEnter(el, done) {
            gsap.fromTo(el, {
                background: '#000'
            }, {
                duration: 1,
                background: '#fff',
                onComplete: done
            })
        }
    }
```
